tests/testthat/test-tags.r

```diff
@@ -893,3 +893,68 @@ test_that("extractPreserveChunks works for emoji strings", {
     c('chunk2', 'chunk1')
   )
 })
+
+
+
+test_that("html render method", {
+  # Have a place holder div and return a span instead
+  obj <- div("example", .render = function(x) {
+    x$name <- "span"
+    x
+  })
+  expect_equal(obj$name, "div")
+  expect_equal(tagify(obj)$name, "span")
+
+  # Add a class to the tag
+  spanExtra <- tagAddRenderFunction(obj, function(x) {
+    tagAppendAttributes(x, class = "extra")
+  })
+  expect_equal(spanExtra$name, "div")
+  expect_equal(spanExtra$attribs$class, NULL)
+  expect_equal(tagify(spanExtra)$name, "span")
+  expect_equal(tagify(spanExtra)$attribs$class, "extra")
+
+  # Replace the previous render method
+  # Should print a `div` with class `"extra"`
+  divExtra <- tagAddRenderFunction(obj, replace = TRUE, function(x) {
+    tagAppendAttributes(x, class = "extra")
+  })
+  expect_equal(divExtra$attribs$class, NULL)
+  expect_equal(tagify(divExtra)$name, "div")
+  expect_equal(tagify(divExtra)$attribs$class, "extra")
+
```

Comment:
It would be good to also add a test where another layer of render function is added with `tagAddRenderFunction()`.